tests/release/pipelines/fbc_release.go

```diff
@@ -15,186 +15,272 @@ import (
+			})
+		})
 	})
 })
+
+func assertBuildPipelineRunCreated(devFw framework.Framework, devNamespace, managedNamespace, fbcAppName string, component *appservice.Component) {
+	Expect(devFw.AsKubeDeveloper.HasController.WaitForComponentPipelineToBeFinished(component, "", devFw.AsKubeDeveloper.TektonController, &has.RetryOptions{Retries: 3, Always: true})).To(Succeed())
+	_, err := devFw.AsKubeDeveloper.HasController.GetComponentPipelineRun(component.Name, fbcAppName, devNamespace, "")
+	Expect(err).ShouldNot(HaveOccurred())
+
+}
+
+func assertReleasePipelineRunSucceeded(devFw, managedFw framework.Framework, devNamespace, managedNamespace, fbcAppName string, component *appservice.Component) {
+	buildPr, err := devFw.AsKubeDeveloper.HasController.GetComponentPipelineRun(component.Name, fbcAppName, devNamespace, "")
+	Expect(err).ShouldNot(HaveOccurred())
+	snapshot, err := devFw.AsKubeDeveloper.IntegrationController.GetSnapshot("", buildPr.Name, "", devNamespace)
+	Expect(err).ShouldNot(HaveOccurred())
+	releaseCR, err := devFw.AsKubeDeveloper.ReleaseController.GetRelease("", snapshot.Name, devNamespace)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	Eventually(func() error {
+		releasePr, err := managedFw.AsKubeAdmin.ReleaseController.GetPipelineRunInNamespace(managedFw.UserNamespace, releaseCR.GetName(), releaseCR.GetNamespace())
+		Expect(err).ShouldNot(HaveOccurred())
+		if !releasePr.IsDone() {
+			return fmt.Errorf("release pipelinerun %s in namespace %s did not finish yet", releasePr.Name, releasePr.Namespace)
+		}
+		GinkgoWriter.Println("Release PR: ", releasePr.Name)
+		Expect(tekton.HasPipelineRunSucceeded(releasePr)).To(BeTrue(), fmt.Sprintf("release pipelinerun %s/%s did not succeed", releasePr.GetNamespace(), releasePr.GetName()))
+		return nil
+	}, releasePipelineRunCompletionTimeout, defaultInterval).Should(Succeed(), "timed out when waiting for release pipelinerun to succeed")
+}
+
+func assertReleaseCRSucceeded(devFw framework.Framework, devNamespace, managedNamespace, fbcAppName string, component *appservice.Component) {
+	Eventually(func() error {
+		buildPr, err := devFw.AsKubeDeveloper.HasController.GetComponentPipelineRun(component.Name, fbcAppName, devNamespace, "")
+		Expect(err).ShouldNot(HaveOccurred())
+		snapshot, err := devFw.AsKubeDeveloper.IntegrationController.GetSnapshot("", buildPr.Name , "", devNamespace)
+		Expect(err).ShouldNot(HaveOccurred())
+		releaseCR, err := devFw.AsKubeDeveloper.ReleaseController.GetRelease("", snapshot.Name, devNamespace)
+		if err != nil {
+			return err
+		}
+		GinkgoWriter.Println("Release CR: ", releaseCR.Name)
+		if !releaseCR.IsReleased() {
+			return fmt.Errorf("release %s/%s is not marked as finished yet", releaseCR.GetNamespace(), releaseCR.GetName())
+		}
+		return nil
+	}, releaseCreationTimeout, defaultInterval).Should(Succeed())
+}
+
+func CreateComponent(devFw framework.Framework, devNamespace, managedNamespace, fbcAppName, fbcCompName string) *appservice.Component {
```

Comment:
should this be extracted somewhere else? It doesn't seem fbc generic and would be nice to use in our other tests

Reply:
I think it can be extracted and used in release suite, not sure if other suites need it.

Reply:
sounds good to me

Reply:
I create a releaseLib.go file in the release directory and more functions may be added in the future.  Any comment for it?

Reply:
I will open a new thread for it